### PR TITLE
Docker Storage Support for the Firmware Manager

### PR DESCRIPTION
**Pull request by dmccoystephenson**
## Problem
The Firmware Manager addon currently only supports storing firmware files in GCP. In a deployment where this isn't desired, we need another way to store the firmware files.

## Solution
Support for storing firmware files in a directory mounted as a docker volume has been added to the project.

## Testing
- Unit tests pass with these changes.
- This has been deployed on WYDOT's Test VM.
- Program starts up correctly & checks for firmware upgrades as expected.
- Initiating a firmware upgrade for Trihydro's Test Bed RSU was successful.

### CODE REVIEW

**Review by mwodahl — APPROVED** [review]
This looks good to me!

File: sample.env [hunk 1]
```diff
@@ -87,9 +87,12 @@ MAPBOX_INIT_LONGITUDE="-104.9903"
 MAPBOX_INIT_ZOOM="10"
 
 # Firmware Manager
-BLOB_STORAGE_PROVIDER=GCP
-BLOB_STORAGE_BUCKET=
+BLOB_STORAGE_PROVIDER=DOCKER
```
Comment (payneBrandon) [inline sample.env:91]: Is this now either GCP or Docker? Should we indicate that somewhere?

File: services/addons/images/firmware_manager/README.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ Available REST endpoints:
 
 To properly run the firmware_manager microservice the following services are also required:
 
-- Cloud based blob storage
+- Cloud based blob storage (if not using a directory mounted as a docker volume)
```
Comment (payneBrandon) [inline services/addons/images/firmware_manager/README.md]: This is a bit confusing, should the Docker volume be organized the same?

File: services/addons/images/firmware_manager/sample.env [hunk 3]
```diff
@@ -7,8 +7,12 @@ PG_DB_USER=""
...
 
 # Blob storage variables
-BLOB_STORAGE_PROVIDER="GCP"
-BLOB_STORAGE_BUCKET=""
+BLOB_STORAGE_PROVIDER=DOCKER
+# GCP Project and Bucket for BLOB storage (if using GCP)
+GCP_PROJECT=
+BLOB_STORAGE_BUCKET=
+# Docker volume mount point for BLOB storage (if using Docker)
+HOST_BLOB_STORAGE_DIRECTORY=./local_blob_storage
```
Comment (payneBrandon) [inline services/addons/images/firmware_manager/sample.env:15]: This is fine to use relative path? Maybe let's indicate that more explicitly in the README or here

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a couple questions, looks good!

File: sample.env (hunk 1 above)
Comment (dmccoystephenson) [reply sample.env:91]: This has been modified to make the supported options more apparent.

File: services/addons/images/firmware_manager/README.md (hunk 2 above)
Comment (dmccoystephenson) [reply services/addons/images/firmware_manager/README.md]: Yes it should. This has been modified to improve clarity.

File: services/addons/images/firmware_manager/sample.env (hunk 3 above)
Comment (dmccoystephenson) [reply services/addons/images/firmware_manager/sample.env:15]: Yes, a relative path can be used. The README has been modified to state that explicitly.

**Comment by dmccoystephenson** [comment]
This PR is ready for re-review. @payneBrandon

**Review by payneBrandon — APPROVED** [review]
lgtm!

**Pull request merged**